htdocs/PI/write/PIWriteRequest.php

```diff
@@ -569,6 +569,17 @@ private function checkUserAuthenticated () {
       }
     }
 
+    private function checkAuthorisation (Site $siteService, \Site $site, $identifier, $indentifierType) {
+      try {
+        $siteService->checkAuthroisedAPIIDentifier($site, $identifier, $indentifierType);
```

Comment:
I've opened an issue around the naming of this function, but you didn't touch it here, so no need to fix it now.